src/biome/text/loggers.py

```diff
@@ -0,0 +1,125 @@
+from typing import Any, Dict, Optional
+
+from allennlp.training import EpochCallback, GradientDescentTrainer
+from mlflow.tracking import MlflowClient
+
+from biome.text.data import InstancesDataset
+from biome.text.training_results import TrainingResults
+
+
+class BaseTrainLogger(EpochCallback):
+    """Base train logger for pipeline training"""
+
+    def init_train(
+        self,
+        pipeline: "Pipeline",
+        trainer_configuration: "TrainerConfiguration",
+        training: InstancesDataset,
+        validation: Optional[InstancesDataset] = None,
+        test: Optional[InstancesDataset] = None,
+    ):
+        """
+        Init train logging
+
+        Parameters
+        ----------
+        pipeline:
+            The training pipeline
+        trainer_configuration:
+            The trainer configuration
+        training:
+            Training dataset
+        validation:
+            Validation dataset
+        test:
+            Test dataset
+        """
+        pass
+
+    def end_train(self, results: TrainingResults):
+        """
+        End train logging
+
+        Parameters
+        ----------
+        results:
+            The training result set
+
+        """
+        pass
+
+    def log_epoch_metrics(self, epoch: int, metrics: Dict[str, Any]):
+        """
+        Log epoch metrics
+
+        Parameters
+        ----------
+        epoch:
+            The current epoch
+        metrics:
+            The metrics related to current epoch
+        """
+        pass
+
+    def __call__(
+        self,
+        trainer: GradientDescentTrainer,
+        metrics: Dict[str, Any],
+        epoch: int,
+        is_master: bool,
+    ):
+        if epoch >= 0:
+            self.log_epoch_metrics(epoch, metrics)
+
+
+class MlflowLogger(BaseTrainLogger):
+    """A common mlflow logger for pipeline training"""
+
+    def __init__(self, experiment_name: str = None, artifact_location: str = None):
+        self._client = MlflowClient()
```

Comment:
Yes, tracking URI is always configurable by env var. For now, I prefer keep this behaviour for all mlflow dependent components, since could be common to all of them in the same project